From dd5ad88f4180e4025d0249d8585f7f5561497e1d Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 15 Jan 2020 12:23:09 +0100
Subject: GCC: do not instantiate CrossThreadCopier template for abstract class
 MediaStreamVideoTrack

This is invalid in GCC. It does not allow instantiating a template for a class that is
abstract.

Bug: 819294
Change-Id: I9047ef62303356e3e47d350cc5b1fa724459a505
---
 .../peerconnection/adapters/web_rtc_cross_thread_copier.h   | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h b/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h
index 54cb7d13bff3..fe6f47b76b42 100644
--- a/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h
+++ b/third_party/blink/renderer/modules/peerconnection/adapters/web_rtc_cross_thread_copier.h
@@ -164,12 +164,6 @@ struct CrossThreadCopier<blink::MediaStreamVideoTrack>
   STATIC_ONLY(CrossThreadCopier);
 };
 
-template <>
-struct CrossThreadCopier<webrtc::VideoTrackInterface>
-    : public CrossThreadCopierPassThrough<webrtc::VideoTrackInterface> {
-  STATIC_ONLY(CrossThreadCopier);
-};
-
 }  // namespace WTF
 
 #endif  // THIRD_PARTY_BLINK_RENDERER_MODULES_PEERCONNECTION_ADAPTERS_WEB_RTC_CROSS_THREAD_COPIER_H_
-- 
2.20.1

